public/index.html

```diff
@@ -24,36 +24,32 @@ <h1>Rollup code-splitting demo</h1>
 		<div data-used-by='both'></div>
 	</div>
 
-	<!-- for modern browsers -->
-	<script type='module'>
-		import('/module/main-a.js');
-		import('/module/main-b.js');
-		window.hasDynamicImport = true;
-	</script>
-
-	<!-- for older browsers -->
-	<script nomodule src='https://unpkg.com/systemjs@0.21.0/dist/system-production.js'></script>
-	<script nomodule>
-		System.import('nomodule/main-a.js');
-		System.import('nomodule/main-b.js');
-	</script>
-
-	<!-- this snippet ensures nomodule fallback for browsers supporting modules but not dynamic import -->
-	<script type="module">
-		if (!window.hasDynamicImport) {
-			const noms = Array.prototype.filter.call(document.getElementsByTagName('script'), x => x.hasAttribute('nomodule'));
-			function nextLoad (nom, s) {
-				if (!(nom = noms.shift())) return;
-				s = document.createElement('script');
-				if (nom.src)
-					s.src = nom.src, s.addEventListener('load', nextLoad), s.addEventListener('error', nextLoad);
-				else
-					s.innerHTML = nom.innerHTML;
-				document.head.appendChild(s);
-				if (!nom.src) nextLoad();
-			}
-			nextLoad();
-		}
-	</script>
+    <script nomodule>
+	    // If a browser does not support modules, feature detection may not not work properly below, so note this here
+		window.loadEntryPoints = null;
+    </script>
+    <script>
+        if (window.loadEntryPoints !== null) {
+            try {
+                // this will throw a runtime exception in browsers that support "import" but not "import()"
+                window.loadEntryPoints = new Function('import("/module/main-a.js");import("/module/main-b.js");');
```

Comment:
We need to support environments with CSP evaluation disabled in the default workflow.

Reply:
Not too experienced in these things. Will a CSP already prevent the successful parsing or just the actual invocation?

Reply:
So many sites (like for example GitHub) disable all use of "eval" with a CSP policy. So that `new Function('console.log("test")'` and `eval('asdf')` will throw and not do any evaluation, or parsing.

This is why SystemJS uses a global callback in its module format, and doesn't just have a JSON / text format or something like that.

So the approach cannot have `new Function(...)` or `eval(...)` in use at all to not restrict these use cases.

Reply:
Also, can a CSP create issues for the dynamic creation of script tags? This would be a real problem, also for the previous approach.

Reply:
Yes - dynamically populating script via innerHTML is not supported in CSP as that is an evaluation as well.

Reply:
So there is no way to add a script dynamically at runtime and current Firefox will not work when a CSP is in place.

Reply:
Note that having the nomodule script as a script with a src would avoid the CSP issue in the original approach.

Reply:
Oh, but I am not using innerHTML. So this could be fine. Do you know some resources about this topic?

Reply:
See for example - https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/script-src#Unsafe_eval_expressions.

Reply:
Thanks!